dox/Hunter.md

```diff
@@ -0,0 +1,155 @@
+ possibilities include toggling on optional code, or selecting from among 
+ several libraries.  With the options declared we find our dependencies.  Each
+ dependency needs to first be run through Hunter using `hunter_add_package`
+ before the usual CMake `find_package` command can be invoked.  With all 
+ dependencies found we tell CMake to descend into our source directory, and then
+ into the testing directory (assuming `BUILD_TESTS` evaluates to `TRUE`).  
+ That's the end of the top-level `CMakeLists.txt`.
+ 
+ The CMake files within the source and testing directory are pure CMake.  
+ Typically this requires a large amount of boilerplate (particularly for 
+ installing targets) so we have defined the following CMake functions (they 
+ will be in scope thanks to the `include` line):
+ 
+ - `add_nwx_library(${my_target} "${SRCS}" "${INCLUDES}")`
+   - This function will create a CMake target called `${my_target}` whose source
+     files are given by the list `"${SRCS}"` and has public header files given
+     by `"${INCLUDES}"`.
+   - The result of this call is a legitimate CMake target that will be properly
+     installed.  You are free to continue setting it up by including things 
+     like the targets it must link against (make sure to include your 
+     dependencies!!!)
+ - `add_catch_cxx_tests(${test_target} "${test_srcs}" ${target2test})`
+   - This function will create a CTest unit test called `${test_target}` that
+     uses the Catch testing framework.  The resulting executable will be built
+     from the sources in the list `"${test_srcs}"` and will test the target
+     library whose CMake target is `${"target2test}`  
+   - This function will automatically add Catch as a dependency of your project.
+   - This assumes the following is present in only one test's source file.
+      ```.cpp
+      #define CATCH_CONFIG_MAIN
+      #include <catch.hpp>
+      ```
+### The `cmake/Hunter/config.cmake` File
+
+We mentioned above that if your project had local Hunter options it needs to 
+contain a `cmake/Hunter/config.cmake` file; for brevity we refer to this file
+as the `config.cmake` file for the remainder of this section.  The 
+`config.cmake` file can be used to set a variety of per dependency options.  As 
+the Hunter manual is a bit difficult to read sometimes, explanations are given 
+below for some of the more common uses.
+
+#### Non-release dependency
+
+Like most package managers, Hunter relies on a versioning system to keep 
+packages straight.  Particularly during development it is common to not have
+any formal releases.  This does not play well with Hunter's default 
+dependency mechanism.  Instead one includes the following in their 
+`config.cmake` file:
+
+```.cmake
+hunter_config(
+        ${dependency_name}
+        VERSION ${version}
+        URL "${gh_repo}?access_token=${token}"
+        SHA1 "42446742165e643a10778f4a3402a5164728d0bb"
```

Comment:
The SHA1 will need to be updated anytime there's a change to the master branch of any dependency we pull in this way, right?

Reply:
Where "in this way" means, pulling from some branch rather than a release version.

Reply:
Correct.  The long-term goal is to base it off a release because then the hash won't change.